tfx/orchestration/experimental/optimizations/beam_fusion.py

```diff
@@ -0,0 +1,197 @@
+# Lint as: python2, python3
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TFX BeamFusionOptimizer definition."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from typing import List
+
+from tfx.components.base import base_node
+from tfx.components.base import base_executor
+from tfx.orchestration.pipeline import Pipeline
+
+class BeamFusionOptimizer(object):
```

Comment:
I think for now we should keep it as a class to be safe, since we still have to add additional graph logic to replace subgraphs with a `FusedComponent` of some sort and there may be other API changes needed. @charlesccychen Do you have an opinion on this?